File: frontend/src/api/authApi.ts
import type { AuthResponse, Credentials } from '../types/auth'
import { api, setAccessToken } from './client'

export const authApi = {
  async register(credentials: Credentials): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/api/auth/register', credentials)
    setAccessToken(response.token)
    return response
  },

  async login(credentials: Credentials): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/api/auth/login', credentials)
    setAccessToken(response.token)
    return response
  },

  /** Revokes the refresh family server-side; the cookie is cleared by the response. */
  async logout(): Promise<void> {
    try {
      await api.post<void>('/api/auth/logout')
    } finally {
      setAccessToken(null)
    }
  },
}

File: mobile/lib/api/auth_api.dart
import 'api_client.dart';

class AuthApi {
  AuthApi(this._client);
  final ApiClient _client;

  Future<String> register(String email, String password) =>
      _authenticate('/api/auth/register', email, password);

  Future<String> login(String email, String password) =>
      _authenticate('/api/auth/login', email, password);

  Future<String> _authenticate(String path, String email, String password) async {
    final json = await _client.request('POST', path,
        body: {'email': email, 'password': password}) as Map<String, dynamic>;
    await _client.saveSession(
      access: json['token'] as String,
      refresh: json['refreshToken'] as String?,
    );
    return json['email'] as String;
  }

  Future<void> logout() async {
    try {
      await _client.request('POST', '/api/auth/logout');
    } finally {
      await _client.clearSession();
    }
  }
}
